ansible/roles-infra/infra-ec2-template-generate/tasks/validate_instances_amis.yaml

```diff
@@ -0,0 +1,116 @@
+  - name: test 1
+    debug: msg="{{ instances }}"
+
+  - name: prep facts
+    set_fact:
+      instances_DoD: "{{ [] }}"  # DoD
+  
+  - name: fail if I can't map an image to aws_ami_region_mapping
+    loop: "{{ instances }}"
+    when:
+    - item.image is defined
+    - aws_ami_region_mapping[aws_region][item.image] is not defined
+    fail:
+      msg: >- 
+        instance.image "{{ item.image }}" is not defined in aws_ami_region_mapping.
+
+  - name: create instances_DoD from instances LoD
+    loop: "{{ instances }}"
+    loop_control:
+      extended: yes
+    set_fact: 
+      instances_DoD: >-
+        {{ 
+          instances_DoD| combine(
+            { ansible_loop.index0: instances[ansible_loop.index0]  }
+          )
+        }}
+
+  - name: test if bump RHEL<VERSION> images to GOLD, if available in aws_ami_region_mapping
+    loop: "{{ instances_DoD|dict2items }}"
+    when: 
+      - item.value['image'] is defined
+      - item.value['image'] is regex("^RHEL.*$")
+      - item.value['image'] is not regex("^.*GOLD$")      
+      - aws_ami_region_mapping[aws_region][item.value['image']|regex_replace("^(.*)$", "\\1GOLD")] is defined
```

Comment:
So `item.value['image']` would simply become `item.image`  is you're using a copy of `instances`.

Not 100% sure but this line with key evaluation could cause error in ansible.

A better way to test presence of a key is:


```yaml
item.image | regex_replace("^(.*)$", "\\1GOLD") in  aws_ami_region_mapping[aws_region] | default([])   
```

Reply:
seems legit.  I think I didn't test that.

Reply:
I'm changing to that test format.  If I used `|default([])`, don't I always resolve True?